### PR TITLE
Tests/hu core 18 dark mode

### LINKED ISSUE #169: TEST-CORE-18 - Dark mode toggle end-to-end test report
# TEST-CORE-0018 - Dark mode toggle end-to-end test report

**Domain:** Core

## User Story

As an authenticated ITESO user, I want to switch between light and dark themes so that I can use the platform comfortably in low-light environments and according to my visual preference.

## Descripción detallada

This test report covers end-to-end and edge-case scenarios for the dark mode toggle implemented in issue #142 (HU-CORE-18). The feature uses `next-themes`, stores preference in `localStorage`, respects the OS `prefers-color-scheme` on first visit, and relies on CSS variables and Tailwind `dark` class.

Reference issue: https://github.com/ReUseITESO/ReUse/issues/142

## Acceptance Criteria

- Theme toggle visible in navbar
- Clicking the toggle switches between light and dark modes
- Preference persisted in `localStorage`
- Initial theme respects OS `prefers-color-scheme`
- No flash of incorrect theme on page load (no FOUC)
- All listed pages render correctly in both themes
- Toggle is keyboard accessible

### Backend

No backend changes. Tests do not require API access.

### Frontend

- The `ThemeToggle` component in `src/components/ui/` and `ThemeProvider` in `src/app/layout.tsx` are the focus.
- Pages to validate: dashboard, communities, auth/verify, auth/check-email, navbar and any shared components.

### Database

Not applicable.

## Testing Notes

This report defines exhaustive Playwright test scenarios for MCP (Modular Code Playground) style E2E testing using Playwright test runner.

Test strategy:
- Automated E2E tests with Playwright covering UI, persistence, OS preference emulation, accessibility, and FOUC prevention.
- Component-level visual checks (optional snapshots) for pages listed in acceptance criteria.

## Test Scenarios (Playwright)

1) Toggle basic functionality
   - Navigate to home (authenticated state if required).
   - Verify `ThemeToggle` exists in navbar.
   - Click toggle once: expect `document.documentElement` (or body) has `.dark` class (or theme changed visually).
   - Click toggle again: expect `.dark` removed.

2) Persistence across reloads
   - Set theme to dark via toggle.
   - Reload page.
   - Expect theme remains dark and `localStorage` key (e.g., `theme`) equals `dark`.

3) Initial OS preference respected
   - Start a fresh session with no `localStorage` theme key.
   - Emulate `prefers-color-scheme: dark` via Playwright context emulation.
   - Load page and expect initial theme to be dark (no toggle click).
   - Repeat for `prefers-color-scheme: light`.

4) No FOUC on load
   - Ensure implementation applies theme server-side/client early (ThemeProvider SSR config).
   - Reload and measure if any flash of wrong theme appears (visual snapshot before/after load or check for body class present immediately).

5) Page rendering correctness (for each target page)
   - For each page in acceptance criteria (dashboard, communities, verify, check-email, etc.):
     - Load page in light theme: take visual snapshot or check critical elements have expected CSS variables (background, text color).
     - Switch to dark theme: take snapshot or check CSS variables / class presence.
     - Compare snapshots or assert element computed styles differ appropriately and no elements use hardcoded colors (spot-check classes like `bg-white`, `text-gray-700`).

6) Accessibility & keyboard
   - Tab to toggle, press Enter/Space, expect theme change.
   - Verify toggle has ARIA label and role.

7) Edge cases
   - localStorage unavailable (simulate disabled storage): ensure toggle still works for session (in-memory) and does not crash.
   - Multiple tabs: change theme in one tab, ensure other tab updates (using `storage` event) or at least does not throw.
   - Browser that does not support `prefers-color-scheme`: default to light.

## Test Data Required

- No user data required. If authentication is required to see navbar, use a fixture account or mock auth.

## Potential Mocks

- Mock authenticated session if tests require logged-in navbar.

## Dependencies & Duplication Check

**Dependencies:** HU-CORE-05 (navbar existence)

**Possible duplicates:** None found; linked PR #143 implements feature.

## Assumptions

- Theme key used in `localStorage` is `theme` (adjust test to real key if different).
- `next-themes` defaults are used and `dark` class is applied to `html` or `body`.

## ⚠️ Warnings

- If components still contain hardcoded color classes, visual assertions may fail—this requires frontend fixes as part of the same story.

### PR DESCRIPTION
**Pull request by victortelles**
## 📌 Summary
This PR adds end-to-end coverage for HU-CORE-18 dark mode toggle behavior in frontend using Playwright.

Implemented and validated scenarios include:
- Light/Dark toggle behavior from navbar button
- Theme persistence after reload (`localStorage`)
- System preference on first visit (`prefers-color-scheme`)
- Keyboard accessibility (Enter)
- Cross-tab synchronization
- Error-path resilience when `localStorage` fails for `theme`

Result: no functional defects were detected for this user story during MCP Playwright execution and local Playwright runs.

---

## 🔍 Type of Change
Mark the relevant option(s):

- [x] Feature (new functionality)
- [ ] Bug fix
- [ ] Agent (development tooling)
- [ ] Documentation
- [ ] Refactor / cleanup
- [ ] CI / DevOps

---

## 🧩 Scope
Which part of the project is affected?

- [ ] Backend
- [x] Frontend
- [ ] Agents
- [ ] Docs
- [ ] CI / Infrastructure

---

## 🤖 Agent Details (if applicable)
If this PR adds or updates an agent:

- **Agent name:** Not applicable
- **Purpose of the agent:** Not applicable
- **Input(s):** Not applicable
- **Output(s):** Not applicable
- **Example included:**
  - [ ] Yes
  - [x] No (explain why)

This PR does not add or modify an agent.

---

## 🧪 Testing
Describe how this change was tested:
v
- [x] Local execution
- [x] Manual testing
- [ ] Unit tests
- [ ] Not applicable (explain why)

Commands executed:
- `npx playwright test tests/core/18-dark-mode.spec.ts --project=chromium`
- `npm run test:e2e`
- `npm run test:e2e:ui`

Outcome:
- 18-dark-mode spec passed.
- No product defects identified for TEST-CORE-18 / HU-CORE-18.

---

## 📎 Related Context
Link to related issues, discussions, or decisions (if any).

- https://github.com/ReUseITESO/ReUse/issues/169
- https://github.com/ReUseITESO/ReUse/issues/142
- https://github.com/ReUseITESO/ReUse/pull/143

---

## ✅ Checklist
Before requesting a review, confirm that:

- [x] Code builds and runs locally
- [x] Changes are small and focused
- [x] Code follows agreed standards
- [ ] Documentation was updated if needed
- [x] No unrelated changes are included
- [x] AI-generated content was reviewed and understood

---

## 📝 Notes for Reviewers
- No functional defects were detected for dark mode toggle behavior in the tested scenarios.

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -56,3 +56,9 @@ yarn-error.log*
 # typescript
 *.tsbuildinfo
 next-env.d.ts
+
+# Playwright e2e artifacts
```
Comment (Ferreirafc1133) [inline frontend/.gitignore:60]: paths de otro PR, tus tests estan en tests/core/.

**Comment by Ferreirafc1133** [comment]
los tests cubren casos buenos. pero este es el cuarto PR del sprint metiendo playwright.config desde cero (164, 170, 172 y tu) y cada quien con su testDir. ponganse de acuerdo los 4 antes de mergear cualquiera, un solo testDir y un solo config.

File: frontend/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { defineConfig, devices } from '@playwright/test';
+
+export default defineConfig({
+  testDir: './tests/',
```
Comment (Ferreirafc1133) [inline frontend/playwright.config.ts]: cuarto testDir distinto. habla con el equipo antes.

File: frontend/tests/core/18-dark-mode.spec.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
+import { test, expect, type Page } from '@playwright/test';
```
Comment (Ferreirafc1133) [inline frontend/tests/core/18-dark-mode.spec.ts:1]: archivo con tabs. pasale prettier.

**Comment by Ferreirafc1133** [comment]
@victortelles recordatorio: #164 ya mergeo con testDir en tests/gamification/ y fixtures en tests/gamification/fixtures/. alinea tu estructura a esa, no a tests/ suelta.

**Comment by Ferreirafc1133** [comment]
@victortelles ping. testDir y fixtures deben alinearse con #164 que ya mergeo.

**Comment by victortelles** [comment]
Entendido, pasare los cambios mas recientes para esta rama para resolver conflictos y adaptarme al codigo mergeado de dev.

**Comment by Ferreirafc1133** [comment]
@victortelles tercer ping. alinear estructura a tests/gamification.

**Comment by victortelles** [comment]
Oye @Ferreirafc1133 pero no es mejor dejar configurada la carpeta global en el archivo (playwright.config.ts) **/tests** que la definamos dentro de **testDir: './tests/**

Por que si, sigo con la actual **  testDir: './tests/gamification',** voy a tener que meter las pruebas de core dentro de la carpeta de gamification y presiento que se va a hacer mucha bola si hago eso. 

tests/
 ├── gamification/
 ├──├── / core/

File: frontend/tests/core/18-dark-mode.spec.ts (hunk 3 above)
Comment (victortelles) [reply frontend/tests/core/18-dark-mode.spec.ts:1]: Ya le pase el prettier, lo añado al commit. pero no se sube ninguna modificacion.
image
 
image

**Comment by Ferreirafc1133** [comment]
hiciste media tarea. quitaste el config y moviste el spec, bien. pero lo enterraste en tests/gamification/core/ como si dark mode fuera un challenge de gamification. va en tests/core/. y el archivo sigue con tabs, dos rondas sin abrir prettier.

File: frontend/tests/core/18-dark-mode.spec.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+2. Validar alternancia claro/oscuro y etiqueta accesible del boton.
+3. Validar persistencia tras recarga.
+4. Validar degradacion ante error de localStorage para key "theme".
+5. Validar acceso por teclado (Enter).
+6. Validar preferencia inicial de sistema (dark/light).
+7. Validar sincronizacion de tema entre pestanas.
+*/
+
+async function goToHome(page: Page) {
+	await page.goto(HOME_URL, { waitUntil: 'domcontentloaded' });
```
Comment (Ferreirafc1133) [inline frontend/tests/core/18-dark-mode.spec.ts:17]: sigue con tabs. dije pasale prettier.

**Comment by Ferreirafc1133** [comment]
ya quedo el path. pero los tabs siguen ahi. tercera ronda con el mismo comentario, prettier --write toma 2 segundos.

File: frontend/tests/core/18-dark-mode.spec.ts (hunk 4 above)
Comment (Ferreirafc1133) [inline frontend/tests/core/18-dark-mode.spec.ts:17]: tabs otra vez. prettier.

**Comment by victortelles** [comment]
Checando prettier.
image

Aplicando prettier.
image

checando 
image

añadiendo a git los archivos 
image

(No se suben)

image

**Comment by Ferreirafc1133** [comment]
@victortelles cuarto ping. los tabs estan ahi viendo todo, escuchando todo. abre prettier y libralos.
